src/pages/workspaces/workspace/notebooks/modals/CloudEnvironmentModal.js

```diff
@@ -267,10 +292,11 @@ export const CloudEnvironmentModal = ({
   )
 
   const getToolLaunchClickableProps = toolLabel => {
-    const doesCloudEnvForToolExist = currentRuntimeTool === toolLabel || (currentApp && toolLabel === tools.galaxy.label)
+    const app = currentApp(toolLabel)
+    const doesCloudEnvForToolExist = currentRuntimeTool === toolLabel || app
     // TODO what does cookieReady do? Found it in the galaxy app launch code, is it needed here?
```

Comment:
@kyuksel I would like to understand more about why this cookie is necessary (for the deployment, one has to modify the request header in order for the cookie access to work). I will add this to the Tuesday agenda.

Reply:
I'm guessing it's for `new tab`ing but not sure. Do you know, @jdcanas?